### PR TITLE
Optimize fmaxf etc.

### PR DESCRIPTION
**Pull request by kripken**
The wasm builtins are very similar to the normal libc functions, except that nans are handled differently. Keep the musl nan handling, and otherwise use the builtins. This is ~41 bytes less in each fmaxf etc. function, and is 30% faster on this silly benchmark:
```cpp
#include <math.h>
#include <stdio.h>

int main() {
  union {
    int i;
    float f;
  } u;
  float sum = 0;
  const int N = 20000;
  for (int i = 0; i < N; i++) {
    for (int j = 0; j < N; j++) {
      u.i = ((i << 15) + j + 5085) ^ j ^ (i >> 2);
      sum += fmaxf(u.f, 0.5);
    }
  }
  printf("%.2f\n", sum);
}
```
After the speedup we are about equal with gcc natively.

Verified this does not change the output of our tests on this, and added more test coverage.

cc @sunfishcode

### CODE REVIEW

**Comment by sunfishcode** [comment]
Clever! I'm surprised by the magnitude of the speedup, but I guess that means those `signbit` calls aren't easy to optimize.

File: tests/core/test_floatvars.cpp [hunk 1]
```diff
@@ -11,14 +11,26 @@
 #include <cfloat>
 #include <cmath>
 
+#define TEST(name, x, y)     printf("    " #name "(%.2f, %.2f) => %.2f, %.2f\n", x, y, name(x, y), name(y, x));
+#define TEST_ABS(name, x, y) printf("abs " #name "(%.2f, %.2f) => %.2f, %.2f\n", x, y, abs(name(x, y)), abs(name(y, x)));
+
+// fmin etc. are not specced to be sensitive to negative zero, and LLVM does
+// depend on that for optimizations, so check only the absolute value there
+#define TESTS(name) \
```
Comment (jgravelle-google) [inline tests/core/test_floatvars.cpp]: yay macros that make things easier to read :)

**Comment by kripken** [comment]
I added more tests for negative zero. Wasm and musl do handle it correctly even if the libc spec doesn't require it, so nice to make sure we don't regress that.
